subcommand/server-acl-init/command_test.go

```diff
@@ -297,6 +338,112 @@ func TestRun_Tokens(t *testing.T) {
+			TokenFlag:  "-create-client-token",
+			PolicyName: "client-token-dc2",
+			PolicyDCs:  []string{"dc2"},
+			SecretName: "release-name-consul-client-acl-token",
+			LocalToken: true,
+		},
+		{
+			TokenFlag:  "-create-sync-token",
+			PolicyName: "catalog-sync-token-dc2",
+			PolicyDCs:  []string{"dc2"},
+			SecretName: "release-name-consul-catalog-sync-acl-token",
+			LocalToken: true,
+		},
+		{
+			TokenFlag:  "-create-inject-namespace-token",
+			PolicyName: "connect-inject-token-dc2",
+			PolicyDCs:  []string{"dc2"},
+			SecretName: "release-name-consul-connect-inject-acl-token",
+			LocalToken: true,
+		},
+		{
+			TokenFlag:  "-create-enterprise-license-token",
+			PolicyName: "enterprise-license-token-dc2",
+			PolicyDCs:  []string{"dc2"},
+			SecretName: "release-name-consul-enterprise-license-acl-token",
+			LocalToken: true,
+		},
+		{
+			TokenFlag:  "-create-snapshot-agent-token",
+			PolicyName: "client-snapshot-agent-token-dc2",
+			PolicyDCs:  []string{"dc2"},
+			SecretName: "release-name-consul-client-snapshot-agent-acl-token",
+			LocalToken: true,
+		},
+		{
+			TokenFlag:  "-create-mesh-gateway-token",
+			PolicyName: "mesh-gateway-token-dc2",
+			PolicyDCs:  nil,
+			SecretName: "release-name-consul-mesh-gateway-acl-token",
+			LocalToken: false,
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.TokenFlag, func(t *testing.T) {
+			bootToken := "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee"
+			tokenFile, fileCleanup := writeTempFile(t, bootToken)
+			defer fileCleanup()
+
+			k8s, consul, cleanup := completeReplicatedSetup(t, resourcePrefix, bootToken)
+			defer cleanup()
+
+			// Run the command.
+			ui := cli.NewMockUi()
+			cmd := Command{
+				UI:        ui,
+				clientset: k8s,
+			}
+			cmd.init()
+			cmdArgs := []string{
+				"-k8s-namespace=" + ns,
+				"-expected-replicas=1",
+				"-acl-replication-token-file", tokenFile,
+				"-enable-acl-replication",
+				"-server-label-selector=component=server,app=consul,release=" + releaseName,
+				"-resource-prefix=" + resourcePrefix,
+				c.TokenFlag,
+			}
+			responseCode := cmd.Run(cmdArgs)
+			require.Equal(t, 0, responseCode, ui.ErrorWriter.String())
+
+			// Check that the expected policy was created.
+			retry.Run(t, func(r *retry.R) {
+				policy := policyExists(r, c.PolicyName, consul)
+				require.Equal(r, c.PolicyDCs, policy.Datacenters)
+			})
+
+			retry.Run(t, func(r *retry.R) {
+				// Test that the token was created as a Kubernetes Secret.
+				tokenSecret, err := k8s.CoreV1().Secrets(ns).Get(c.SecretName, metav1.GetOptions{})
+				require.NoError(r, err)
+				require.NotNil(r, tokenSecret)
+				token, ok := tokenSecret.Data["token"]
+				require.True(r, ok)
+
+				// Test that the token has the expected policies in Consul.
+				tokenData, _, err := consul.ACL().TokenReadSelf(&api.QueryOptions{Token: string(token)})
+				require.NoError(r, err)
+				require.Equal(r, c.PolicyName, tokenData.Policies[0].Name)
+				require.Equal(r, c.LocalToken, tokenData.Local)
```

Comment:
We can only get policy name and ID here because we only have an ACLLink object:
```go
type ACLLink struct {
	ID   string
	Name string
}
```
We do check PolicyDC right above this and the two are correlated by name.

Reply:
Oh, I see it in a retry block above. I must have missed that when reviewing it the first time.